ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -1435,22 +1430,22 @@ public List<IndexEntry> getValidEntry(IndexSegment indexSegment, FileSpan duplic
       copyCandidates.removeIf(copyCandidate ->
           isDuplicate(copyCandidate, duplicateSearchSpan, indexSegment.getStartOffset(), checkAlreadyCopied) || (
               config.storeContainerDeletionEnabled && isFromDeprecatedContainer(copyCandidate)));
-      if (duplicateSearchSpan != null && validEntriesSize == copyCandidates.size()
-          && config.storeCompactionEnableBasicInfoOnMissingDuplicate) {
```

Comment:
QQ: It seems like previously we always use the default value which is false. And since it has no error messages pops out, I feel no one will pay attention to this error unless someone enable trace log on purpose.

Reply:
The configuration was created for a specific reason back then. We saw some Put IndexEntries having duplicates. Later we figured out it's because we have a list of log segments under compaction but they are not consecutive. After we fixed that this never happened again.